app/components/combined_system_message/index.js

```diff
@@ -3,21 +3,61 @@
 
 import {connect} from 'react-redux';
 import {bindActionCreators} from 'redux';
+import {createSelector} from 'reselect';
 
 import {getProfilesByIds, getProfilesByUsernames} from 'mattermost-redux/actions/users';
 import {Preferences} from 'mattermost-redux/constants';
 import {getBool, getTeammateNameDisplaySetting} from 'mattermost-redux/selectors/entities/preferences';
-import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
+import {getCurrentUser, getProfiles} from 'mattermost-redux/selectors/entities/users';
 
 import CombinedSystemMessage from './combined_system_message';
 
-function mapStateToProps(state) {
+const getUserProfiles = createSelector(
+    getProfiles,
+    (state, props) => props.allUserIds,
+    (state, props) => props.allUsernames,
+    (profiles, allUserIds, allUsernames) => {
+        const allProfiles = profiles.reduce((acc, profile) => {
```

Comment:
Instead of making a new map with both users by username and by id, you could use the `getProfiles` and `getUsersByUsername` selectors separately since you know which map to look at for each one.